dor/providers/process_basic_image.py

```diff
@@ -0,0 +1,212 @@
+import hashlib
+import shutil
+import tempfile
+import uuid
+from datetime import UTC, datetime
+from pathlib import Path
+from typing import Callable
+
+from dor.adapters.generate_service_variant import generate_service_variant, ServiceImageProcessingError
+from dor.adapters.make_intermediate_file import make_intermediate_file
+from dor.adapters.technical_metadata import (
+    ImageMimetype, TechnicalMetadata, TechnicalMetadataError, get_technical_metadata
+)
+from dor.providers.file_system_file_provider import FilesystemFileProvider
+from dor.builders.parts import FileInfo, MetadataFileInfo, UseFunction, UseFormat, flatten_use
+from dor.providers.models import (
+    Agent, AlternateIdentifier, FileReference, PackageResource, FileMetadata, PreservationEvent
+)
+from dor.providers.serializers import PreservationEventSerializer
+from dor.providers.utilities import sanitize_basename
+from dor.settings import template_env
+
+
+ACCEPTED_IMAGE_MIMETYPES = [
+    ImageMimetype.JPEG,
+    ImageMimetype.TIFF,
+    ImageMimetype.JP2
+]
+
+
+class FileSetIdentifier:
+
+    def __init__(self, project_id: str, file_name: str):
+        self.project_id = project_id
+        self.file_name = file_name
+        self.basename = sanitize_basename(Path(file_name).stem)
+
+    @property
+    def alternate_identifier(self) -> str:
+        return f"{self.project_id}:{self.basename}"
+
+    @property
+    def uuid(self) -> uuid.UUID:
+        hex_string = hashlib.md5(self.alternate_identifier.encode("UTF-8")).hexdigest()
+        return uuid.UUID(hex=hex_string)
+
+    @property
+    def identifier(self) -> str:
+        return str(self.uuid)
+
+
+def create_preservation_event(
+    type: str,
+    collection_manager_email: str,
+    detail: str = ""
+):
+    event = PreservationEvent(
+        identifier=str(uuid.uuid4()),
+        type=type,
+        datetime=datetime.now(tz=UTC),
+        detail=detail,
+        agent=Agent(
+            role="image_processing",
+            address=collection_manager_email
+        )
+    )
+    return event
+
+
+def convert_metadata_file_info_to_file_metadata(metadata_file_info: MetadataFileInfo) -> FileMetadata:
+    return FileMetadata(
+        id=metadata_file_info.xmlid,
+        use=flatten_use(*metadata_file_info.uses),
+        ref=FileReference(
+            locref=metadata_file_info.locref,
+            mimetype=metadata_file_info.mimetype,
+            mdtype=metadata_file_info.mdtype
+        )
+    )
+
+
+def create_file_set_descriptor_file(
+    resource: PackageResource,
+    descriptor_file_path: Path
+) -> None:
+
+    entity_template = template_env.get_template("preservation_mets.xml")
+    xmldata = entity_template.render(
+        resource=resource,
+        object_identifier=resource.id,
+        create_date=datetime.now().strftime("%Y-%m-%dT%H:%M:%SZ"),
+    )
+    with (descriptor_file_path).open("w") as file:
+        file.write(xmldata)
```

Comment:
Could separate out the side effect (file writing), which could make it easier to test when/if we get to that.